### PR TITLE
Added keyframes-name-pattern rule

### LINKED ISSUE #3262: Add keyframes-name-pattern
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I'd like to make an Feature request, where `@keyframes` definition's name needs to match given pattern, something like `selector-class-pattern` rule, but for keyframes.

I'd like to use an namespace validation rule for all my keyframe definitions.

> Which rule, if any, is this issue related to?

Issue relates to similar rule `selector-class-pattern`.

> What CSS is needed to reproduce this issue?

```css
@keyframes NAMESPACE-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

n/a

> Which version of stylelint are you using?

`9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config myconfig *.css` should be ok.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

I expect warnings to be flagged if keyframe definition name doesn't match the pattern.

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing related to the feature request, because of lacking rules.

### PR DESCRIPTION
**Pull request by MatthiasKunnen**
> Which issue, if any, is this issue related to?

Closes #3262

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/keyframes-name-pattern/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      actual: pattern,
+      possible: [_.isString]
+    });
+
+    if (!validOptions) {
+      return;
+    }
+
+    const regex = _.isString(pattern) ? new RegExp(pattern) : pattern;
+    root.walkAtRules(/^keyframes/i, keyframesNode => {
```
Comment (evilebottnawi) [inline lib/rules/keyframes-name-pattern/index.js]: Not all `keyframes` starts with `keyframes`, we have already prefixed version `--webkit-keyframes`, need add supports and tests. Thanks for work!

**Comment by MatthiasKunnen** [comment]
@evilebottnawi, the failing check seems to be something that needs addressing outside of this PR. Can you confirm?

**Comment by MatthiasKunnen** [comment]
Thank you, let me know if there is any more feedback on this PR.

**Comment by ntwb** [comment]
Added to changelog in https://github.com/stylelint/stylelint/commit/1b152cacb1747ccfa69e6cdd1a5d5c9810130640

> Added: `keyframes-name-pattern` rule (#3321).

**Pull request merged**